balancer/weightedroundrobin/balancer.go

```diff
@@ -339,16 +393,21 @@ type picker struct {
 	v         uint32             // incrementing value used by the scheduler; accessed atomically
 	cfg       *lbConfig          // active config when picker created
 	subConns  []*weightedSubConn // all READY subconns
+
+	// These fields are built out at creation time and read only after, so no
+	// need for synchronization.
+	target          string
+	locality        string
+	metricsRecorder estats.MetricsRecorder
 }
 
-// scWeights returns a slice containing the weights from p.subConns in the same
-// order as p.subConns.
 func (p *picker) scWeights() []float64 {
```

Comment:
This should take the boolean and pass it through, probably, since it will cause metrics to be recorded, but doesn't seem like it would from its name.